synapse/handlers/auth.py

```diff
@@ -626,6 +629,10 @@ def validate_login(self, username, login_submission):
         # special case to check for "password" for the check_password interface
         # for the auth providers
         password = login_submission.get("password")
+
+        if password and PY2:
+            password = password.decode('utf8')
```

Comment:
why is this necessary? surely login_submission is coming from the json body so should already have been un-utf8'ed?

(and if it is necessary, why don't we do it on PY3?)

Reply:
(should it not just be `password = unicode(password)` ?)

Reply:
`unicode(password)` also will decode it with ASCII, as it's forced decoding.